backend/src/main/service/CachingService.ts

```diff
@@ -0,0 +1,71 @@
+import { RedisClient } from 'redis';
+import { CachingObject } from '../data/CachingObject';
+import { Promotion } from '../entity/Promotion';
+
+export class CachingService {
+  private client: RedisClient;
+
+  constructor(client: RedisClient) {
+    this.client = client;
+  }
+
+  async cacheLatLonValues(
```

Comment:
Could we add documentation for each of the functions here? And also one that summarizes the class?